bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BulkReadVeneerApi.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.wrappers.veneer;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.core.InternalApi;
+import com.google.api.gax.batching.Batcher;
+import com.google.bigtable.v2.ReadRowsRequest;
+import com.google.bigtable.v2.RowFilter;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.internal.RequestContext;
+import com.google.cloud.bigtable.data.v2.models.Filters;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.hbase.adapters.Adapters;
+import com.google.cloud.bigtable.hbase.wrappers.BulkReadWrapper;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.hbase.client.Result;
+
+/** For internal use only - public for technical reasons. */
+@InternalApi("For internal usage only")
+public class BulkReadVeneerApi implements BulkReadWrapper {
+
+  private static final BlockingQueue<RowFilter> BATCHER_QUEUE = new LinkedBlockingQueue<>();
+
+  private final ScheduledExecutorService batcherPool;
```

Comment:
I think that the executor pool should be static as well.

I'm thinking something along the lines of:

```java
  private static BlockingQueue BATCHER_CLEANER_QUEUE = ....;
  private static Thread BATCHER_CLEANER = new Thread(daemon=true, new Runner() {
    public void run() {
      try {
        while(true) {
          BATCHER_CLEANER_QUEUE.next().close();
        }
      catch(InterruptedException e) {
         // 
      }
    }
});

 static {
   jvm.addShutdownHook(new Runner() {
      BATCHER_CLEANER.interrupt();
   });
}
```

Some improvements to the above sketch:
- maybe put the initialization logic into a static method that gets called on the first construction of a BulkRead
- make the shutdown hook close all of the batchers that the cleaner thread didn't finish
- add a todo to remove this hack once veneer supports an asyncClose